ingress/controllers/nginx/utils.go

```diff
@@ -109,6 +109,15 @@ func getLBDetails(kubeClient *unversioned.Client) (*lbInfo, error) {
 		return nil, fmt.Errorf("Unable to get POD information")
 	}
 
+	if pod.Status.Phase != api.PodRunning {
+		// we wait up to 30 seconds until the pod is running and
+		// it is possible to get the IP and name of the node
+		err := waitForPodRunning(kubeClient, podNs, podName, time.Millisecond*200, time.Second*30)
```

Comment:
@bprashanth because of this https://github.com/kubernetes/contrib/issues/752. 
For some reason `kubeClient.Nodes().Get(pod.Spec.NodeName)` returns an error unless I add this wait. All this is required just to get the IP of the node (to update the ingress rules).